src/pages/Game.js

```diff
@@ -208,6 +208,17 @@ export default class Game extends Component {
     return lastMessage > this.state.lastReadChat;
   }
 
+  get userColorKey() {
+    return `user_color_${window.location.href}`;
+  }
+
+  get userColor() {
+    let color =
```

Comment:
The previous logic for rendering the user cursor determined the user color in a weird way (thereby denormalizing the source of truth for user color). Your refactor solves that problem by making the user cursor color derive primarily from `this.game.users` -- great! Is the localStorage part necessary?  Seems like `this.game.users[this.props.id]?.color` will be truthy in almost all cases after the initial load.